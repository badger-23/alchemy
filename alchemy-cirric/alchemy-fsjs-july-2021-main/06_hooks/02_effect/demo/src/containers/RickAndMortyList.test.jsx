import React from 'react';
import { render, screen } from '@testing-library/react';
import RickAndMortyList from './RickAndMortyList';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import listData from '../fixtures/list.json';

const server = setupServer(
  rest.get('https://rickandmortyapi.com/api/character/', (req, res, ctx) => {
    return res(ctx.json(listData));
  })
);

describe('RickAndMortyList Container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays a list of characters on the page', async () => {
    const { container } = render(
      <MemoryRouter>
        <RickAndMortyList />
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
