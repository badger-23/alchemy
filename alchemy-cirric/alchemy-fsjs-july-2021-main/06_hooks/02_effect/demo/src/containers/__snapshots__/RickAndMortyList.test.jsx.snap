// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyList Container displays a list of characters on the page 1`] = `
<div>
  <ul
    aria-label="characters"
  >
    <li>
      <a
        href="/1"
      >
        <figure>
          <img
            alt="Rick Sanchez"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <figcaption>
            Rick Sanchez
             - 
            Human
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/2"
      >
        <figure>
          <img
            alt="Morty Smith"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <figcaption>
            Morty Smith
             - 
            Human
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/3"
      >
        <figure>
          <img
            alt="Summer Smith"
            src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
          />
          <figcaption>
            Summer Smith
             - 
            Human
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/4"
      >
        <figure>
          <img
            alt="Beth Smith"
            src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
          />
          <figcaption>
            Beth Smith
             - 
            Human
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/5"
      >
        <figure>
          <img
            alt="Jerry Smith"
            src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
          />
          <figcaption>
            Jerry Smith
             - 
            Human
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/6"
      >
        <figure>
          <img
            alt="Abadango Cluster Princess"
            src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
          />
          <figcaption>
            Abadango Cluster Princess
             - 
            Alien
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/7"
      >
        <figure>
          <img
            alt="Abradolf Lincler"
            src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
          />
          <figcaption>
            Abradolf Lincler
             - 
            Human
             - 
            unknown
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/8"
      >
        <figure>
          <img
            alt="Adjudicator Rick"
            src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
          />
          <figcaption>
            Adjudicator Rick
             - 
            Human
             - 
            Dead
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/9"
      >
        <figure>
          <img
            alt="Agency Director"
            src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
          />
          <figcaption>
            Agency Director
             - 
            Human
             - 
            Dead
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/10"
      >
        <figure>
          <img
            alt="Alan Rails"
            src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
          />
          <figcaption>
            Alan Rails
             - 
            Human
             - 
            Dead
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/11"
      >
        <figure>
          <img
            alt="Albert Einstein"
            src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
          />
          <figcaption>
            Albert Einstein
             - 
            Human
             - 
            Dead
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/12"
      >
        <figure>
          <img
            alt="Alexander"
            src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
          />
          <figcaption>
            Alexander
             - 
            Human
             - 
            Dead
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/13"
      >
        <figure>
          <img
            alt="Alien Googah"
            src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
          />
          <figcaption>
            Alien Googah
             - 
            Alien
             - 
            unknown
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/14"
      >
        <figure>
          <img
            alt="Alien Morty"
            src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
          />
          <figcaption>
            Alien Morty
             - 
            Alien
             - 
            unknown
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/15"
      >
        <figure>
          <img
            alt="Alien Rick"
            src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
          />
          <figcaption>
            Alien Rick
             - 
            Alien
             - 
            unknown
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/16"
      >
        <figure>
          <img
            alt="Amish Cyborg"
            src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
          />
          <figcaption>
            Amish Cyborg
             - 
            Alien
             - 
            Dead
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/17"
      >
        <figure>
          <img
            alt="Annie"
            src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
          />
          <figcaption>
            Annie
             - 
            Human
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/18"
      >
        <figure>
          <img
            alt="Antenna Morty"
            src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
          />
          <figcaption>
            Antenna Morty
             - 
            Human
             - 
            Alive
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/19"
      >
        <figure>
          <img
            alt="Antenna Rick"
            src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
          />
          <figcaption>
            Antenna Rick
             - 
            Human
             - 
            unknown
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/20"
      >
        <figure>
          <img
            alt="Ants in my Eyes Johnson"
            src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
          />
          <figcaption>
            Ants in my Eyes Johnson
             - 
            Human
             - 
            unknown
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
